### PR TITLE
Add uci_defaults.txt: quick intro + common pitfall

### PR DESCRIPTION
**Pull request by Borromini**
Provides a quick explanation of how uci-defaults works, and the importance of implementing checks if you don't want custom settings overwritten by accident.

### CODE REVIEW

**Comment by jow-** [comment]
I think this should got into a new subtree, something like `Scripting` or `Development`. It certainly does not fit among the uci config file descriptions imho

**Comment by Borromini** [comment]
You have a point there. So Documentation > Development?

Which could then include buildroot explanation and stuff, or is that too much? Your earlier point about not duplicating info between these pages and a wiki is still very valid but i think for some aspects (like these) it's good to have them documented on the main site (which in time might refer to the wiki for more in-depth explanations e.g.). Especially since stuff like buildroot, UCI etc. is not in flux as much as support status on devices, or howtos etc.

**Comment by blogic** [comment]
i was thinking of a "features" section that describes all these little mechanisms that exist. we could also have a short text about preinit, snaphots, extroot, ....

File: docs/uci_defaults.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+============
+
+== Integrating custom settings through UCI defaults
+
+LEDE allows you to add custom settings by adding scripts into the 'files/etc/uci-defaults/' directory, which
+will then be run after flashing. When keeping settings, this should mean the scripts are run *after* the upgrade
+process flashed the image and appended the saved settings to the JFFS partition (which will be mounted as 
+'/overlay'). The path is identical for the buildroot and the image generator. Scripts should not be executable.
+To ensure your scripts are not interfering with any other scripts, make sure they get executed last by giving them
+a high prefix (e.g. 99_customisations). A basic script could look like this:
```
Comment (karlp) [inline docs/uci_defaults.txt:13]: this isn't actually true.  I've raised it before iirc.  They're alphabetical, but package names are after numbers, so zzzzzzzzzzz_yourscriptname is more reliable than 99_yourscript.  You can do something like https://github.com/remakeelectric/openwrt-cc/commit/9f83c196763641f1e11e7f8cc3712e9adc43f3f8 to see the real order as they progress.

File: docs/uci_defaults.txt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+uci -q batch <<-EOT
+  add dhcp host
+  set dchp.@host[0].name='bellerophon'
+  set network.@host[0].ip='192.168.2.100'
+  set network.@host[0].mac='a1:b2:c3:d4:e5:f6'
+    EOT
+---
+
+Once the script has run successfully and exited cleanly, it will be removed from '/etc/uci-defaults/'. You can
```
Comment (karlp) [inline docs/uci_defaults.txt:27]: might be worth explicitly stating what "exited cleanly" means?

Comment (Borromini) [inline docs/uci_defaults.txt:27]: I can change that to 'exits with status 0'?

**Comment by Borromini** [comment]
I assume the best way is to close this pull request and generate a new one, so I'll do so.

**Pull request closed without merging**